src/test/java/io/managed/services/test/kafka/QuotaKafkaInstanceTest.java

```diff
@@ -0,0 +1,204 @@
+    private static final String KAFKA_INSTANCE_NAME_NO_QUOTA = "mk-e2e-no-quota-" + Environment.LAUNCH_KEY;
+    private static final String KAFKA_INSTANCE_NAME_FAIL = "mk-e2e-quota-fail-" + Environment.LAUNCH_KEY;
+
+    // Kafka instance plans
+    private static final String PLAN_DEVELOPER = "developer.x1";
+    private static final String PLAN_STANDARD = "standard.x1";
+    
+    // Kafka API code errors
+    private static final String KAFKAS_MGMT_120_CODE = "KAFKAS-MGMT-120";
+    private static final String KAFKAS_MGMT_120_REASON = "Insufficient quota: error getting billing model: No available billing model found";
+    private static final String KAFKAS_MGMT_21_CODE = "KAFKAS-MGMT-21";
+    private static final String KAFKAS_MGMT_21_REASON = "unable to detect instance type in plan provided: ";
+
+    private KafkaMgmtApi quotaUserKafkaMgmtApi;
+    private KafkaMgmtApi noQuotaUserKafkaMgmtApi;
+    
+    private KafkaRequest kafka;
+
+    @BeforeClass
+    public void bootstrap() {
+        assertNotNull(Environment.DIFF_ORG_USERNAME, "the DIFF_ORG_USERNAME env is null");
+        assertNotNull(Environment.DIFF_ORG_PASSWORD, "the DIFF_ORG_PASSWORD env is null");
+        assertNotNull(Environment.ALIEN_USERNAME, "the ALIENUSERNAME env is null");
+        assertNotNull(Environment.ALIEN_PASSWORD, "the ALIENPASSWORD env is null");
+
+        var quotaUserAuth = new KeycloakLoginSession(Environment.DIFF_ORG_USERNAME, Environment.DIFF_ORG_PASSWORD);
+        var noQuotaUserAuth = new KeycloakLoginSession(Environment.ALIEN_USERNAME, Environment.ALIEN_PASSWORD);
+
+        ApplicationServicesApi quotaUserApi = ApplicationServicesApi.applicationServicesApi(quotaUserAuth);
+        ApplicationServicesApi noQuotaUserApi = ApplicationServicesApi.applicationServicesApi(noQuotaUserAuth);
+
+        this.quotaUserKafkaMgmtApi = quotaUserApi.kafkaMgmt();
+        this.noQuotaUserKafkaMgmtApi = noQuotaUserApi.kafkaMgmt();
+        
+        deleteExistingKafkaInstances();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown() {
+        deleteExistingKafkaInstances();
+    }
+
+    private void deleteExistingKafkaInstances() {
+        assumeTeardown();
+        
+        try {
+            KafkaMgmtApiUtils.cleanKafkaInstance(quotaUserKafkaMgmtApi, KAFKA_INSTANCE_NAME_QUOTA);
+        } catch (Throwable t) {
+            LOGGER.error("failed to clean kafka instance: ", t);
+        }
+        
+        try {
+            KafkaMgmtApiUtils.cleanKafkaInstance(noQuotaUserKafkaMgmtApi, KAFKA_INSTANCE_NAME_NO_QUOTA);
+        } catch (Throwable t) {
+            LOGGER.error("failed to clean kafka instance: ", t);
+        }
+
+        try {
+            KafkaMgmtApiUtils.cleanKafkaInstance(noQuotaUserKafkaMgmtApi, KAFKA_INSTANCE_NAME_FAIL);
+        } catch (Throwable t) {
+            LOGGER.error("failed to clean kafka instance: ", t);
+        }
+    }
```

Comment:
in current implementaiton (of `testQuotaUserSucceededCreateStandardInstance `) we would also need to add 
```
try {
            KafkaMgmtApiUtils.cleanKafkaInstance(quotaUserKafkaMgmtApi, KAFKA_INSTANCE_NAME_FAIL);
```            
**why:** assuming scenario first instance isn't created correctly (failed test, not created instance) later
`testQuotaUserFailedCreateStandardInstanceWhenQuotaIsReached` would create instance (another failed test) and because we are not clearing it  (we are deleting instance with given name but with "wrong" user) would be stucked here till manual deletion.